### ISSUE TITLE
Segmentation fault when unnamed variable in negated record

### ISSUE THREAD

**Issue by taipan-snake** [souffle-lang/souffle#420]
A rule of the following form causes a segfault in souffle:
`a([x]) :- b([x]), !c([x, _]).`

An example program with this issue is:
```
.type r = [x:number, y:number]
.type s = [x:number]

.decl a(x:r)
a([2, 3]).

.decl b(x:s)
b([1]).

.decl c(x:s)
c([x]) :- b([x]), !a([x, _]).

.output c
```

The equivalent program without records:
```
.decl a(x:number, y:number)
a(2, 3).

.decl b(x:number)
b(1).

.decl c(x:number)
c(x) :- b(x), !a(x, _).

.output c
```
runs fine.

**Comment by b-scholz** [comment]
A semantic check is required. 

Record arguments need to be grounded when used in constraints (like negation).

**Comment by olligobber** [comment]
Fixed in 82b7edefed2832c8b5454747025733f99dd5d573
